src/ops/fuelup_toolchain/export.rs

```diff
@@ -0,0 +1,320 @@
+    if let Some(path) = get_fuel_toolchain_toml() {
+        toolchain_info_path = path;
+        if force {
+            println_warning(&format!(
+                "Because the `--force` argument was supplied, the toolchain info file at {} will be removed.",
+                &toolchain_info_path.display(),
+            ));
+            fs::remove_file(&toolchain_info_path).unwrap();
+        } else {
+            println_warning(&format!(
+                "There is an existing toolchain info file at {}. \
+                Do you wish to replace it with a new one? (y/N) ",
+                &toolchain_info_path.display(),
+            ));
+            let mut need_replace = String::new();
+            reader.read_line(&mut need_replace).unwrap();
+            if need_replace.trim() == "y" {
+                fs::remove_file(&toolchain_info_path).unwrap();
+            } else {
+                bail!(
+                    "Failed to create a new toolchain info file at {} \
+                    because a toolchain info file already exists at that location.",
+                    &toolchain_info_path.display(),
+                );
+            }
+        }
+    }
+    let mut toolchain_name = Toolchain::from_settings()?.name;
+    if let Some(name) = name {
+        toolchain_name = name;
+    }
+
+    let export_toolchain = match DistToolchainDescription::from_str(&toolchain_name) {
+        Ok(desc) => Toolchain::from_path(&desc.to_string()),
+        Err(_) => Toolchain::from_path(&toolchain_name),
+    };
+
+    let mut version_map: HashMap<String, Version> = HashMap::new();
+    for component in Components::collect_exclude_plugins()? {
+        let component_executable = export_toolchain.bin_path.join(&component.name);
+        if let Ok(version) = exec_version(component_executable.as_path()) {
+            version_map.insert(component.name.clone(), version);
+        };
+
+        if component.name == component::FORC {
+            let forc_executables = component.executables;
+            for plugin in Components::collect_plugins()? {
+                if !forc_executables.contains(&plugin.name) {
+                    if !plugin.is_main_executable() {
+                        for executable in plugin.executables.iter() {
+                            let plugin_executable = export_toolchain.bin_path.join(executable);
+                            if let Ok(version) = exec_version(plugin_executable.as_path()) {
+                                version_map.insert(executable.clone(), version);
+                            };
+                        }
+                    } else {
+                        let plugin_executable = export_toolchain.bin_path.join(&plugin.name);
+                        if let Ok(version) = exec_version(plugin_executable.as_path()) {
+                            version_map.insert(plugin.name.clone(), version);
+                        };
+                    }
```

Comment:
I don't think we don't need to check plugin.is_main_executable() because the behavior is the same.

```suggestion
                    plugin.executables.iter().for_each(|executable| {
                        exec_version(export_toolchain.bin_path.join(executable))
                            .map(|version| version_map.insert(executable.clone(), version));
                    });
```